crates/client/data-availability/src/near/mod.rs

```diff
@@ -0,0 +1,77 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use near_da_primitives::Blob;
+use near_da_rpc::near::config::KeyType;
+use near_da_rpc::near::Client;
+use near_da_rpc::DataAvailability;
+use tokio::sync::RwLock;
+
+use crate::{DaClient, DaMode};
+
+pub mod config;
+use config::NearConfig;
+
+#[derive(Clone)]
+pub struct NearClient {
+    client: Arc<Client>,
+    last_published_txid: Arc<RwLock<Option<String>>>,
+    mode: DaMode,
+}
+
+impl NearClient {
+    pub fn new_blocking(config: NearConfig) -> Result<Self> {
+        let client_config = near_da_rpc::near::config::Config {
+            key: KeyType::SecretKey(config.account_id, config.secret_key),
+            contract: config.contract_id,
+            network: config.network,
+            namespace: config.namespace,
+        };
+
+        Ok(Self {
+            client: Arc::new(Client::new(&client_config)),
+            last_published_txid: Arc::new(RwLock::new(None)),
+            mode: config.mode,
+        })
+    }
+}
+
+#[async_trait]
+impl DaClient for NearClient {
+    fn get_mode(&self) -> DaMode {
+        self.mode
+    }
+
+    async fn publish_state_diff(&self, state_diff: bytes::Bytes) -> Result<()> {
+        // setter
+
+        let res = self
+            .client
+            .submit(&[Blob::new_v0(self.client.config.namespace, state_diff.to_vec())])
+            .await
+            .map_err(|e| anyhow::anyhow!("failed to submit blobs: {e}"))?;
```

Comment:
do we need to wait to check of the txn was finally inlcuded in the block?

Reply:
no, `submit` submits and waits for finality

Reply:
if possible, can you add a comment for this?